maps/randomvaults/objects.dm

```diff
@@ -630,3 +636,213 @@
 
 		CO.face_atom(AM)
 		CO.visible_message("<span class='notice'>\The [CO] looks at \the [AM] and hisses angrily!</span>")
+
+
+/* Spy Sat stuff */
+
+/obj/machinery/power/magtape_deck
+	name = "magnetic tape drive"
+	desc = "A primitive way of storing information. Used because of its longevity over most digital counterparts."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "blackbox"
+
+	active_power_usage = 500
+	density = 1
+	var/last_check
+
+/obj/machinery/power/magtape_deck/New()
+	..()
+	connect_to_network()
```

Comment:
Does this not get done automatically?

Either way, even if it does, I'm not sure `New()` is a reliable place for this.

Reply:
It is not done automatically from what I can find. Treadmills do it, battery_ports do it. Normally it is done in wrenchAnchor event, but as this starts anchored, it should automatically connect (Like the treadmill does) so it has a cable to check.